SchoolApplication/School.Repository/StudentRepository.cs

```diff
@@ -14,123 +14,177 @@ public class StudentRepository : BaseRepository<Student>
+                                                        select sc.StudentId).Count()
+                                    }).ToList();
+                    return students;
+                    //used before to get Coursecount without Jooin
+                    //var StdList = db.Students.ToList();
+
+                    //foreach (var student in StdList)
+                    //{
+                    //  StudentDtoPost dto = new StudentDtoPost();
+                    //   dto.student = student;
+                    //    dto.CoursesCount = db.StudentCourses.Where(x => x.StudentId == student.Id).Count();
+
+                    //    DtoList.Add(dto);
+
+                    // }
+
+
+                }
+            }
+            catch (Exception e)
+            {
+
+                throw e;
+            }
+        
+
+
+        }
+        public StudentDtoPost FindStudent(int id)
+        {
+            try
+            {
+                
+                using (var context = new SchoolContext())
                 {
-                    var StdList=db.Students.ToList();
-                    foreach (var student in StdList)
-                    {
-                        StudentDtoPost dto = new StudentDtoPost();
-                        dto.student = student;
-                        dto.CoursesCount = db.StudentCourses.Where(x => x.StudentId == student.Id).Count();
+                   var students = (from s in context.Students.Where(x => x.Id == id)
+                                    select new StudentDtoPost
+                                    {
+                                        student = s,
+                                        StudentCourses=(from sc in context.StudentCourses
+                                                        join c in context.Courses on sc.CourseId equals c.CourseId
+                                                        where sc.StudentId==id
+                                                        select new CourseDTO()
+                                                        {
+                                                            CourseId=sc.CourseId,
+                                                            Name=c.Name
+                                                        }
+                                                        ).ToList()
+                                    }).FirstOrDefault();
+                    return students;
+                }
+            }
+            catch (Exception e)
+            {
 
-                        DtoList.Add(dto);
-                        
+                throw e;
+            }
+        }
+        public bool AddStudent(Student obj, String courses)
+        {
+            try
+            {
+                using (SchoolContext db = new SchoolContext())
+                {
+                    StudentCourse stdCourse = new StudentCourse();
+                    db.Students.Add(obj);
```

Comment:
obj ? I think It would be better approach to use student rather than obj for more understandablity